src/cargo/ops/cargo_new.rs

```diff
@@ -89,6 +89,12 @@ fn get_name<'a>(path: &'a Path, opts: &'a NewOptions, config: &Config) -> CargoR
 }
 
 fn check_name(name: &str) -> CargoResult<()> {
+    if name == "test" {
+        bail!("The name `{}` cannot be used as a crate name\n\
```

Comment:
stylistically Cargo error messages tend to start with a lowercase letter (like the compiler) rather than an uppercase letter